src/python/foglamp/core/scheduler.py

```diff
@@ -898,22 +918,20 @@ def _schedule_first_task(self, schedule, current_time):
         elif schedule.type == Schedule.Type.STARTUP:
             schedule_execution.next_start_time = current_time
 
-        self._logger.info(
-            "Scheduled task for schedule '%s' to start at %s", schedule.name,
-            datetime.datetime.fromtimestamp(schedule_execution.next_start_time))
+        if self._logger.isEnabledFor(logging.INFO):
+            self._logger.info(
```

Comment:
why this check is needed; it will only log info if its enable setup to that level

Reply:
It's to avoid an expensive call to datetime.datetime.fromtimestamp unnecessarily